src/gscreenshot/__init__.py

```diff
@@ -37,7 +37,24 @@ def __init__(self, screenshooter=None):
 
         self.saved_last_image = False
         self.last_save_file = None
-        self.last_save_directory = os.path.expanduser("~")
+        if os.path.isfile(self.get_save_file()):
+            f = open(self.get_save_file(), "r")
+            self.last_save_directory = f.readline()
+            f.close()
+        else:
+            self.last_save_directory = os.path.expanduser("~")
+            self.save_last_directory(self.last_save_directory)
+
+    def get_save_file(self):
+        if 'XDG_CACHE_HOME' in os.environ:
+            return os.environ['XDG_CACHE_HOME'] + "/gscreenshot"
+        else:
+            return os.path.expanduser("~/.gscreenshot")
```

Comment:
~/.cache/gscreenshot might be a better spot for this. I generally don't like dropping hidden files right in the home directory.

Reply:
That's what the `if 'XDG_CACHE_HOME' in os.environ` is there for, if user has environment variable XDG_CACHE_HOME (usually "~/.cache", but can be changed in /etc/profile), it'll create the configuration in user's cache directory. However if the user doesn't have the XDG variable defined, I think it's better to just drop it in user's home directory as that is the standard approach

Reply:
Ah! Got it, I've never used the XDG_CACHE_HOME variable before. Makes sense to me.